cpp/katz_centrality_module/algorithm/katz.hpp

```diff
@@ -0,0 +1,131 @@
+#include <algorithm>
+#include <cmath>
+#include <map>
+#include <set>
+
+#include <mg_exceptions.hpp>
+#include <mg_graph.hpp>
+
+namespace katz_alg {
+namespace {
+
+class KatzCentralityData;
+
+///
+///@brief Calculates maximum degree in graph. Degree is denoted by number of edges going from node.
+///
+///@param graph Graph for calculation
+///@return std::uint64_t
+///
+std::uint64_t MaxDegree(const mg_graph::GraphView<> &graph);
+
+///
+///@brief Calculates whether the upper and lower boundaries within the list of active nodes are epsilon-separated.
+///
+///@param active_nodes Active nodes for convergence calculations
+///@param k Factor for sorting only first k
+///@param epsilon Factor of separation/tolerance
+///@return true
+///@return false
+///
+bool Converged(std::set<std::uint64_t> &active_nodes, std::uint64_t k, double epsilon);
+
+///
+///@brief Running the convergence process of shrinking the upper and lower bound for katz centrality. Updates are stored
+/// in context data structure.
+///
+///@param active_nodes Active nodes - list of nodes in convergence process
+///@param graph Graph for convergence calculation
+///@param alpha Attenuation factor
+///@param k Parameter for keeping the first k node centralities sorted
+///@param epsilon Tolerance factor
+///@param gamma Convergence factor
+///@return std::vector<std::pair<std::uint64_t, double>>
+///
+std::vector<std::pair<std::uint64_t, double>> KatzCentralityLoop(std::set<std::uint64_t> &active_nodes,
```

Comment:
In the rest of your code, the `alpha`, `k` and `epsilon` parameters of `KatzCentralityLoop` are always `katz_alg::alpha` and so on. With this in mind, is it possible to remove these parameters from the signature as `katz_alg::alpha` etc. are visible from inside the function?

Reply:
I tried to hide them in the private methods of this module. We can discuss what is the best way to do it, but it certainly is a valuable proposal :+1:

Reply:
Either way looks good to me :ok_hand: